### ISSUE TITLE
nosql causes exceptions in Node.js > v10.0.0

### ISSUE THREAD

**Issue by seetd** [oauthinaction/oauth-in-action-code#14]
nosql will cause an exception in `FileReader.prototype.open` line 2388 will cause an exception in Node.js > v10.0.0. The callback parameter is no longer optional for `fs.close`. Not passing it will throw a TypeError at runtime. There was a depreciation warning since Node.js v7.0.0. My workaround for now is to change `fs.close` to `fs.closeSync` in line 2388 of the index.js file in node_modules/nosql. The author has indeicated that he will be making changes to the module soon. Refer to petersirka/nosql#46

**Comment by ken5scal** [comment]
I am having the same issue with Node.js v10.12.0.

**Comment by mathiasconradt** [comment]
Yes, had the same. Downgraded to 8.0.0 - or "nvm use 8.0", then it worked for me. Didn't try 10.x.

Related https://github.com/oauthinaction/oauth-in-action-code/issues/18#issuecomment-442599911

**Comment by jricher** [comment]
This code was originally written on a very old version of Node.js. Looks like we might need to run through and update the internal dependencies, thanks for the pointers and confirmations.

**Comment by dmark** [comment]
9.2.0 seems to work as well.

**Comment by dsnkostic** [comment]
If you do not want to change the node version, here is what you can do

* First you need to update `nosql` library to the most recent one (`6.1.0` seems to be working properly). 
Here is how `package.json` should look like:
```
{
  "dependencies": {
    "express": "^4.13.1",
    "sync-request": "^2.0.1",
    "body-parser": "^1.13.2",
    "underscore": "^1.8.3",
    "underscore.string": "^3.1.1",
    "consolidate": "^0.13.1",
    "qs": "^4.0.0",
    "randomstring": "^1.0.7",
    "nosql": "^6.1.0",
    "cors": "^2.7.1"
  }
}
```
Don't forget to execute `npm update` in the folder of the `package.json`
* Go in `protectedResource.js` and update method that is causing the error to this one:

```
nosql.one().make(function(builder) {
  builder.where('access_token', inToken);
  builder.callback(function(err, response) {
    console.log(response);
    if (response) {
      console.log("We found a matching token: %s", inToken);
    } else {
      console.log('No matching token was found.');
    };
    req.access_token = response;
    next();
    return;
  })
});
```

This should fix the issue.

Author should update and retest all of the exercises as this issue makes `oAuth2 In Action` book unusable at the very beginning (unless you want to downgrade your nodejs)

**Comment by simongrigorian** [comment]
I would install NVM, and install node 8 -- this would fix the deprecated issue.

**Comment by timmi4sa** [comment]
> If you do not want to change the node version, here is what you can do
> 
> * First you need to update `nosql` library to the most recent one (`6.1.0` seems to be working properly).
>   Here is how `package.json` should look like:
> 
> ```
> {
>   "dependencies": {
>     "express": "^4.13.1",
>     "sync-request": "^2.0.1",
>     "body-parser": "^1.13.2",
>     "underscore": "^1.8.3",
>     "underscore.string": "^3.1.1",
>     "consolidate": "^0.13.1",
>     "qs": "^4.0.0",
>     "randomstring": "^1.0.7",
>     "nosql": "^6.1.0",
>     "cors": "^2.7.1"
>   }
> }
> ```
> 
> Don't forget to execute `npm update` in the folder of the `package.json`
> 
> * Go in `protectedResource.js` and update method that is causing the error to this one:
> 
> ```
> nosql.one().make(function(builder) {
>   builder.where('access_token', inToken);
>   builder.callback(function(err, response) {
>     console.log(response);
>     if (response) {
>       console.log("We found a matching token: %s", inToken);
>     } else {
>       console.log('No matching token was found.');
>     };
>     req.access_token = response;
>     next();
>     return;
>   })
> });
> ```
> 
> This should fix the issue.
> 
> Author should update and retest all of the exercises as this issue makes `oAuth2 In Action` book unusable at the very beginning (unless you want to downgrade your nodejs)

Is it me or the authorizationServer.js file also requires a conversion from "nosql.all()" to "nosql.all().make(..)"? Otherwise I am getting the same error once updated the "nosql" module to version 6.1.0.

**Comment by jricher** [comment]
Should be fixed by #44
